### ISSUE TITLE
Log4J 2.x support

### ISSUE THREAD

**Issue by mst-appear** [hazelcast/hazelcast#2345]
It would be nice to have Hazelcast log to Log4J 2.x.

Log4J 2.x is not directly compatible with Log4J 1.4, but it comes with an emulation layer to support applications using Log4J 1.4. However, this does not work with Hazelcast. I raised an issue about this with the Log4J team, but they consider it to be an issue with Hazelcast, and suggested that I raise the issue here. See
https://issues.apache.org/jira/browse/LOG4J2-611

So it would be nice if either Log4jFactory can be fixed to work with Log4J 2.x, or if a new LoggerFactory for Log4J 2.x is added.

I found a workaround by using the Slf4jFactory and the SLF4J bridge in Log4J 2.x. However, I would prefer to not depend on SLF4J, since it can cause class loading issues if multiple components using it in the same JVM. (I am already getting there by using Hazelcast in a web app, running that web app in Jetty and trying to get Jetty to use Log4J also.)
